build/tasks/AnalyzeBuildGraph.cs

```diff
@@ -174,6 +181,14 @@ private void EnsureConsistentGraph(IEnumerable<ArtifactInfo.Package> packages, I
                 Log.LogError("Package versions are inconsistent. See build log for details.");
             }
 
+            foreach (var versions in dependencyMap.Values)
+            {
+                foreach (var item in versions.Where(i => !i.IsReferenced))
+                {
+                    Log.LogMessage(MessageImportance.Normal, $"Potentially unused external dependency: {item.PackageId}/{item.Version}");
```

Comment:
Why "potentially"? Also, what should someone do if they see this message?

Reply:
I put this message as "normal" level so it won't normally be seen unless someone is intentionally working on pruning our dependency list.

It says potentially because it's possible, though rare, that a package may be used but not as a PackageReference. This analysis only detects PackageReference. There are a few ways this can happen:
 - implicit package references (like NETStandard.Library and Microsoft.NETCore.App)
 - packages that are nuspec only dependencies, like MVC's dependency on Microsoft.DiaSymReader.Native
 - packages that were just barely added but are not yet referenced in a submodule, like Polly.Extensions.Http-Signed

Reply:
Can we create a wiki page that has this exact info on it, and link to it from the message? Otherwise I'm afraid that whoever sees that message is unlikely to know exactly what it means and how to act on it.

Reply:
https://github.com/aspnet/Universe/wiki/Build-warning-and-error-codes#potentially-unused-external-dependency